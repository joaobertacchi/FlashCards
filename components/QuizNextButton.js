// @flow
import React from 'react';
import { Text, TouchableOpacity } from 'react-native';

type Props = {
  index: number,
  total: number,
  onShowNextQuestion: () => void,
  onShowQuizResult: () => void,
};

const QuizNextButton = ({
  index,
  total,
  onShowNextQuestion,
  onShowQuizResult,
}: Props) => (
  index + 1 < total
    ? (
      <TouchableOpacity onPress={onShowNextQuestion}>
        <Text>Next question</Text>
      </TouchableOpacity>
    )
    : (
      <TouchableOpacity onPress={onShowQuizResult}>
        <Text>Quiz result</Text>
      </TouchableOpacity>
    )
);

export default QuizNextButton;
