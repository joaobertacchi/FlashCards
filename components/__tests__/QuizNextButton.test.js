import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import QuizNextButton from '../QuizNextButton';

describe('QuizNextButton snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders correctly Next Question button', () => {
    const tree = renderer.create(<QuizNextButton index={0} total={2} disabled={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Quiz Result button', () => {
    const tree = renderer.create(<QuizNextButton index={1} total={2} disabled={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Next Question button', () => {
    const tree = renderer.create(<QuizNextButton index={0} total={2} disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly Quiz Result button', () => {
    const tree = renderer.create(<QuizNextButton index={1} total={2} disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
