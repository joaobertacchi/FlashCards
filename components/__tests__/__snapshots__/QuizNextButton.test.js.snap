// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizNextButton snapshot renders correctly Next Question button 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginHorizontal": 20,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2f95dc",
        "borderRadius": 7,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "minWidth": 180,
            "textAlign": "center",
          },
          Object {
            "minWidth": undefined,
          },
        ]
      }
    >
      Next question
    </Text>
  </View>
</View>
`;

exports[`QuizNextButton snapshot renders correctly Next Question button 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginHorizontal": 20,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "lightgrey",
        "borderColor": "grey",
        "borderRadius": 7,
        "borderWidth": 1,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "grey",
            "fontSize": 18,
            "minWidth": 180,
            "textAlign": "center",
          },
          Object {
            "minWidth": undefined,
          },
        ]
      }
    >
      Next question
    </Text>
  </View>
</View>
`;

exports[`QuizNextButton snapshot renders correctly Quiz Result button 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginHorizontal": 20,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#2f95dc",
        "borderRadius": 7,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
            "minWidth": 180,
            "textAlign": "center",
          },
          Object {
            "minWidth": undefined,
          },
        ]
      }
    >
      Quiz result
    </Text>
  </View>
</View>
`;

exports[`QuizNextButton snapshot renders correctly Quiz Result button 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginHorizontal": 20,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "lightgrey",
        "borderColor": "grey",
        "borderRadius": 7,
        "borderWidth": 1,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "grey",
            "fontSize": 18,
            "minWidth": 180,
            "textAlign": "center",
          },
          Object {
            "minWidth": undefined,
          },
        ]
      }
    >
      Quiz result
    </Text>
  </View>
</View>
`;
